src/NuGet.Services.EndToEnd/Support/Clients/V2V3SearchClient.cs

```diff
@@ -0,0 +1,253 @@
+                searchBaseUrls.Add(_testSettings.SearchBaseUrl);
+            }
+            else
+            {
+                searchBaseUrls.AddRange(await _v3IndexClient.GetSearchBaseUrls());
+            }
+
+            var queryUrl = searchBaseUrls
+                .Select(u => $"{u}/query?{queryString}")
+                .ToList().First();
+
+            return await _httpClient.GetJsonAsync<V3SearchResponse>(queryUrl);
+        }
+
+        public async Task<AutocompleteResponse> Autocomplete(string queryString)
+        {
+            var searchBaseUrls = new List<string>();
+            if (_testSettings.SearchBaseUrl != null)
+            {
+                searchBaseUrls.Add(_testSettings.SearchBaseUrl);
+            }
+            else
+            {
+                searchBaseUrls.AddRange(await _v3IndexClient.GetSearchBaseUrls());
+            }
+
+            var queryUrl = searchBaseUrls
+                .Select(u => $"{u}/autocomplete?{queryString}")
+                .ToList().First();
+
+            return await _httpClient.GetJsonAsync<AutocompleteResponse>(queryUrl);
+        }
+
+        /// <summary>
+        /// Polls all V2 search URLs until the provided ID and version are available. If <see cref="TestSettings.SearchBaseUrl"/>,
+        /// is configured, then only that search service is polled. We poll the V2 search URL because it allows us to
+        /// easily ignore all filtering rules and query for a single version.
+        /// </summary>
+        /// <param name="id">The package ID.</param>
+        /// <param name="version">The package version.</param>
+        /// <param name="logger">The logger.</param>
+        /// <returns>Returns a task that completes when the package is available or the timeout has occurred.</returns>
+        public async Task WaitForPackageAsync(string id, string version, ITestOutputHelper logger)
+        {
+            var searchBaseUrls = new List<string>();
+            if (_testSettings.SearchBaseUrl != null)
+            {
+                searchBaseUrls.Add(_testSettings.SearchBaseUrl);
+            }
+            else
+            {
+                searchBaseUrls.AddRange(await _v3IndexClient.GetSearchBaseUrls());
+            }
+
+            var v2SearchUrls = searchBaseUrls
+                .Select(u => $"{u}/search/query")
+                .ToList();
+
+            Assert.True(v2SearchUrls.Count > 0, "At least one search base URL must be configured.");
+
+            logger.WriteLine(
+                $"Waiting for package {id} {version} to be available on search endpoints:" + Environment.NewLine +
+                string.Join(Environment.NewLine, v2SearchUrls.Select(u => $" - {u}")));
+
+            var tasks = v2SearchUrls
+                .Select(u => WaitForPackageAsync(u, id, version, logger))
+                .ToList();
+
+            await Task.WhenAll(tasks);
+        }
+        
+        public async Task<IReadOnlyList<string>> GetSearchBaseUrlsAsync()
```

Comment:
Call this to get the list of search base URLs to test. Then enumerate calling the `Query` method on each. You could add a helper method for this that returns a dictionary where the key is the search base URL and the value is the search result from that specific search service. Whichever is clearer for you.